qemu/tests/block_hotplug.py

```diff
@@ -185,16 +192,32 @@ def get_new_disk(disk1, disk2):
                                   logging.info)
             utils_test.run_virt_sub_test(test, params, env, sub_type)
 
-        for num in xrange(blk_num):
-            error_context.context("Unplug block device (iteration %d)" % iteration,
-                                  logging.info)
-            disks_before_unplug = find_disk(vm, get_disk_cmd)
-            device_list[num].unplug(vm.monitor)
-            device_list[num].verify_unplug("", vm.monitor)
-            unplug_status = utils_misc.wait_for(lambda: len(get_new_disk(find_disk(vm, get_disk_cmd),
-                                                disks_before_unplug)) != 0, pause)
-            if not unplug_status:
-                test.fail("Failed to unplug disks")
+        error_context.context("Unplug block device (iteration %d)" % iteration,
+                              logging.info)
+        if not test_img_list:
+            test_img_list = img_list[1:]
+
+        for image_id in test_img_list:
+            for device in vm.devices:
+                if device.get_param("id") == image_id:
+                    disks_before_unplug = find_disk(vm, get_disk_cmd)
+                    device.unplug(vm.monitor)
+                    device.verify_unplug("", vm.monitor)
+                    unplug_status = utils_misc.wait_for(lambda: len(get_new_disk(find_disk(vm, get_disk_cmd),
+                                                        disks_before_unplug)) != 0, pause)
+                    if not unplug_status:
+                        test.fail("Failed to unplug disks")
+                    if device.get_param("bus"):
+                        device_controller_id = device.get_param("bus").split(".")[0]
+                        device_controller_list.append(device_controller_id)
+                    vm.devices.remove(device)
+                    break
```

Comment:
How many devices will be unplugged ?
I think this might be `continue`

Reply:
device is identified by "id=img_id", ex -device id=stg0, so for every img_id, only one device is unplugged. So the check will executed untill all non-system image_id are traversed.